backend/app/settings.py

```diff
@@ -0,0 +1,182 @@
+
+
+class Environment(str, Enum):
+    PRODUCTION = "production"
+    DEV = "dev"
+    TEST = "test"
+
+
+class GlobalSettings(BaseSettings):
+    LOG_LEVEL: str = logging.getLevelName(logging.INFO)
+    IS_TESTING: bool = False
+    DATABASE_URI: PostgresDsn = "postgresql+psycopg2://ess:ess@localhost:5442/ess"
+    SOLR_URL: AnyUrl = "http://localhost:8983/solr/"
+    RS_URL: AnyUrl = "http://localhost:9080/"
+    RS_ROWS: int = 1000
+    BACKEND_BASE_URL: AnyUrl = "http://localhost:8000"
+    UI_BASE_URL: AnyUrl = "http://localhost:4200"
+    OIDC_HOST: AnyUrl = "https://aai-demo.eosc-portal.eu"
+    OIDC_AAI_NEW_API: bool = False
+    OIDC_CLIENT_ID: str = "NO_CLIENT_ID"
+    OIDC_CLIENT_SECRET: str = "NO_CLIENT_SECRET"
+
+    OIDC_ISSUER: str = f"{OIDC_HOST}/oidc/"
+    OIDC_AUTH_ENDPOINT: str = OIDC_OLD_AUTH_ENDPOINT
+    OIDC_TOKEN_ENDPOINT: str = OIDC_OLD_TOKEN_ENDPOINT
+    OIDC_USERINFO_ENDPOINT: str = OIDC_OLD_USERINFO_ENDPOINT
+    OIDC_JWKS_ENDPOINT = OIDC_OLD_JWKS_ENDPOINT
+
+    STOMP_HOST: str = "127.0.0.1"
+    STOMP_PORT: int = 61613
+    STOMP_LOGIN: str = "guest"
+    STOMP_PASS: str = "guest"
+    STOMP_USER_ACTIONS_TOPIC: str = "/topic/user_actions"
+    STOMP_CLIENT_NAME: str = "dev-client"
+    STOMP_SSL: bool = False
+
+    SHOW_RECOMMENDATIONS: bool = True
+    RECOMMENDER_ENDPOINT: AnyUrl = "http://localhost:8081/recommendations"
+    MARKETPLACE_BASE_URL: AnyUrl = "https://marketplace.eosc-portal.eu"
+    EOSC_COMMONS_URL: AnyUrl = "https://s3.cloud.cyfronet.pl/eosc-portal-common/"
+    EOSC_COMMONS_ENV: str = "production"
+    EOSC_EXPLORE_URL: AnyUrl = "https://explore.eosc-portal.eu"
+    RELATED_SERVICES_ENDPOINT: AnyUrl = "https://beta.providers.eosc-portal.eu/api/public/interoperabilityRecord/relatedResources"
+
+    class Config:
+        fields = {
+            "STOMP_USER_ACTIONS_TOPIC": {"env": "ESS_STOMP_USER_ACTION"},
+            "STOMP_CLIENT_NAME": {"env": "ESS_QUEUE_CLIENT_NAME"},
+            "STOMP_SSL": {"env": "ESS_STOMP_SSL"},
+        }
+
+
+class DevSettings(GlobalSettings):
+    pass
+
+
+class TestSettings(GlobalSettings):
+    IS_TESTING: bool = True
+    DATABASE_URI: PostgresDsn = (
+        "postgresql+psycopg2://ess_test:ess_test@localhost:5452/ess_test"
+    )
+
+
+class ProductionSettings(GlobalSettings):
+    pass
+
+
+SETTINGS_MAP = {
+    "production": ProductionSettings,
+    "dev": DevSettings,
+    "test": TestSettings,
+}
+
+
+def get_settings():
+    """Builds an appropriate config basing on env var ENVIRONMENT value"""
+    environment = Environment(os.environ.get("ENVIRONMENT"))
+    config = SETTINGS_MAP[environment.value]()
+    if config.OIDC_AAI_NEW_API:
+        config.OIDC_ISSUER = f"{config.OIDC_HOST}'/auth/realms/core"
+        config.OIDC_JWKS_ENDPOINT = OIDC_NEW_JWKS_ENDPOINT
+        config.OIDC_USERINFO_ENDPOINT = OIDC_NEW_USERINFO_ENDPOINT
+        config.OIDC_TOKEN_ENDPOINT = OIDC_NEW_TOKEN_ENDPOINT
+        config.OIDC_AUTH_ENDPOINT = OIDC_NEW_AUTH_ENDPOINT
```

Comment:
It would be more elegant to move this code into `GlobalSettings` class itself - this kind of hacking can be done using validators (check `pre=True`)